remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/mappers/JsonRulesMapper.kt

```diff
@@ -223,12 +225,18 @@ private val attributesMappers = mapOf(
     Pair("netpOnboarded", networkProtectionOnboarded),
 )
 
-fun List<JsonMatchingRule>.mapToMatchingRules(): Map<Int, List<MatchingAttribute<*>>> = this.map {
-    Pair(it.id, it.attributes.map { attrs -> attrs.map() })
+fun List<JsonMatchingRule>.mapToMatchingRules(
+    attributeMatchers: Set<JsonToMatchingAttributeMapper>,
+): Map<Int, List<MatchingAttribute<*>>> = this.map {
+    Pair(it.id, it.attributes.map { attrs -> attrs.map(attributeMatchers) })
 }.toMap()
 
-private fun Map.Entry<String, JsonMatchingAttribute>.map(): MatchingAttribute<*> {
+private fun Map.Entry<String, JsonMatchingAttribute>.map(attributeMatchers: Set<JsonToMatchingAttributeMapper>): MatchingAttribute<*> {
     return runCatching {
+        attributeMatchers.forEach {
+            val matchingAttribute = it.map(this.key, this.value)
+            if (matchingAttribute != null) return@runCatching matchingAttribute
+        }
         attributesMappers[this.key]?.invoke(this.value) ?: Unknown(this.value.fallback)
```

Comment:
Notice we keep old mappers and logic. So both (injected mappers and RMF mappers) coexists.